### PR TITLE
Chore: Remove settings Fibers usage

### PR DESCRIPTION
**Pull request by ggazzo**

### CODE REVIEW

File: apps/meteor/app/settings/server/raw.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import Settings from '../../models/server/models/Settings';
+import { Settings } from '@rocket.chat/models';
 
 const cache = new Map();
 
 export const setValue = (_id, value) => cache.set(_id, value);
 
 const setFromDB = async (_id) => {
-	const setting = Settings.findOneById(_id, { fields: { value: 1 } });
+	const setting = await Settings.findOneById(_id, { fields: { value: 1 } });
```
Comment (sampaiodiego) [inline apps/meteor/app/settings/server/raw.js:8]:
```suggestion
	const setting = await Settings.findOneById(_id, { projection: { value: 1 } });
```

File: apps/meteor/app/ui-master/server/index.js [hunk 2]
```diff
@@ -124,16 +125,11 @@ Meteor.startup(() => {
...
-		const colors = Settings.find({ _id: /theme-color-rc/i }, { fields: { value: 1, editor: 1 } })
-			.fetch()
-			.filter((color) => color && color.value);
-
-		if (!colors) {
-			return;
-		}
+		const colors = await Settings.find({ _id: /theme-color-rc/i }, { projection: { value: 1, editor: 1 } }).toArray();
 		const css = colors
+			.filter((color) => color && color.value)
```
Comment (sampaiodiego) [inline apps/meteor/app/ui-master/server/index.js:132]: is it better filter colors without values or let it go to CSS with an empty value?

because if `color` has no `value`, with this `filter` there will be no `--color-name: ` into the final CSS

File: apps/meteor/app/settings/server/SettingsRegistry.ts [hunk 3]
```diff
@@ -83,21 +83,21 @@ const compareSettings = compareSettingsIgnoringKeys([
 ]);
 
 export class SettingsRegistry {
-	private model: typeof SettingsModel;
+	private model: typeof Settings;
```
Comment (sampaiodiego) [inline apps/meteor/app/settings/server/SettingsRegistry.ts:86]:
```suggestion
	private model: ISettingsModel;
```

File: apps/meteor/app/livechat/server/methods/saveIntegration.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
+import { Settings } from '@rocket.chat/models';
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/methods/saveIntegration.ts:1]: should all updates in this file have an `await`?

Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/methods/saveIntegration.ts:1]: should all updates in this file have an `await`?

File: apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import { retrieveRegistrationStatus } from './retrieveRegistrationStatus';
+
+export async function unregisterWorkspace() {
+	const { workspaceRegistered } = retrieveRegistrationStatus();
+	if (!workspaceRegistered) {
+		return true;
+	}
+
+	Settings.updateValueById('Cloud_Workspace_Id', undefined);
```
Comment (sampaiodiego) [inline apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts:11]: why did you change to `undefined`?

Comment (sampaiodiego) [reply apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts:11]: maybe `await Promise.all( ... )`?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/app/ui-master/server/index.js (hunk 2 above)
Comment (ggazzo) [reply apps/meteor/app/ui-master/server/index.js:132]: valid question, but since this is the original filter, I will kindly ignore, since its not the PR scope ;)

File: apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	}
+
+	await Promise.all([
+		Settings.updateValueById('Cloud_Workspace_Id', undefined),
+		Settings.updateValueById('Cloud_Workspace_Name', undefined),
+		Settings.updateValueById('Cloud_Workspace_Client_Id', undefined),
+		Settings.updateValueById('Cloud_Workspace_Client_Secret', undefined),
+		Settings.updateValueById('Cloud_Workspace_Client_Secret_Expires_At', undefined),
+		Settings.updateValueById('Cloud_Workspace_PublicKey', undefined),
+		Settings.updateValueById('Cloud_Workspace_Registration_Client_Uri', undefined),
```
Comment (sampaiodiego) [inline apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts:18]:
```suggestion
		Settings.updateValueById('Cloud_Workspace_Id', null),
		Settings.updateValueById('Cloud_Workspace_Name', null),
		Settings.updateValueById('Cloud_Workspace_Client_Id', null),
		Settings.updateValueById('Cloud_Workspace_Client_Secret', null),
		Settings.updateValueById('Cloud_Workspace_Client_Secret_Expires_At', null),
		Settings.updateValueById('Cloud_Workspace_PublicKey', null),
		Settings.updateValueById('Cloud_Workspace_Registration_Client_Uri', null),
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts (hunk 6 above)
Comment (ggazzo) [reply apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts:18]: this is the ONLY PLACE where we use null instead of undefined, type definitions predict undefined, what is the gain/benefit of using null somewhere?

Comment (sampaiodiego) [reply apps/meteor/app/cloud/server/functions/unregisterWorkspace.ts:18]: I actually found just a SINGLE PLACE using `undefined` 🤔 the is in TS, so it can't anything else.. but clearly there are more usage of `null`..

I don't understand why you trust more on the type definition than on this function.. the point is, `undefined` is ignored and the record is not updated.. so the correct is `null`, the type definition must be changed to reflect this.

**Review by sampaiodiego — APPROVED** [review]
